addons/website_slides/static/src/js/slides_share.js

```diff
@@ -7,31 +7,64 @@ import { _t } from 'web.core';
 var ShareMail = publicWidget.Widget.extend({
     events: {
         'click button': '_sendMail',
+        'keypress input': '_onKeypress',
     },
 
     //--------------------------------------------------------------------------
     // Private
     //--------------------------------------------------------------------------
 
     /**
+     * Send the email(s) on 'Enter' key
+     *
      * @private
+     * @param {Event} ev
      */
+    _onKeypress: function (ev) {
+        if (ev.keyCode === $.ui.keyCode.ENTER) {
+            ev.preventDefault();
+            this._sendMail();
+        }
+    },
+
     _sendMail: function () {
         var self = this;
+        const channelID = this.$('button').data('channel-id');
         var input = this.$('input');
         var slideID = this.$('button').data('slide-id');
-        if (input.val() && input[0].checkValidity()) {
+        let resID;
```

Comment:
Same here, will have to see how this can be factorize since I would like to keep the routes split.
I think it's possible to keep it nice and with not too much copy/pasting, maybe by simply having a method that takes the route as argument? Will have to see.